### ISSUE TITLE
Fetch and http request

### ISSUE THREAD

**Issue by jvailius** [jupyterlite/xeus#80]
A follow up on https://github.com/jupyterlite/pyodide-kernel/issues/96

## Description

I tried xeus because adding additional pkgs seems far easier then in pyodide. But I ran into issues with network requests.

## Reproduce

Using fetch seems to work in principle but gave me this error:

```python
import js
import json

json.loads(await (await js.fetch("http://localhost:8000/jupyter/lab/jupyter-lite.json")).text())
```
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
File /lib/python3.11/site-packages/IPython/__init__.py:44, in patch_asyncio.<locals>.asyncio_call(self, coro)
     41     task.add_done_callback(done_cb)
     43 raw_js_py_callback = pyjs.JsValue(py_callback)
---> 44 js_py_callback = raw_js_py_callback['__usafe_void_val_val__'].bind(raw_js_py_callback)
     45 js_promise = pyjs.js.Promise.new(js_py_callback)
     47 pyjs.js.globalThis.toplevel_promise = js_promise

AttributeError: has no attribute/key __usafe_void_val_val__
```

Then I tried `httpx` which seems to require openssl  (I added it to my environment.yml). This made the kernel unusable with the condole error: `LinkError: WebAssembly.instantiate(): Import #92 module="env" function="time": imported function does not match the expected type`. Complete output below.

I can confirm that `requests` lib works but I could not test it with https so far. 

## Context

- JupyterLite version: 0.2.3
- Operating System and version: debian bookworm
- Browser and version: Chrome, Version 122.0.6261.111 (Official Build) (64-bit)

<details><summary>Browser Output</summary>
<pre>
<div aria-expanded="true" tabindex="-1" style="box-sizing: border-box; min-width: 0px; min-height: 0px; flex: 0 0 auto;"><span class="source-code" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--source-code-font-family); white-space: pre-wrap; font-size: var(--source-code-font-size) !important; line-height: 1.2;"><span class="console-message-anchor" style="box-sizing: border-box; min-width: 0px; min-height: 0px; float: right; text-align: right; max-width: 100%; margin-left: 4px;"><span class="devtools-link" role="link" tabindex="-1" title="http://localhost:8000/jupyter/extensions/@jupyterlite/xeus/static/870.1260b274e8c70c25c3c4.js?v=1260b274e8c70c25c3c4:2" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--sys-color-primary); text-decoration: underline; outline-offset: 2px; cursor: pointer; word-break: break-all;"><br class="Apple-interchange-newline">870.1260b274e8c70c25<span class="devtools-link-ellipsis" style="box-sizing: border-box; min-width: 0px; min-height: 0px;">…</span>0b274e8c70c25c3c4:2</span> </span><span class="console-message-text" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--override-error-text-color) !important;">Uncaught (in promise) <span class="object-value-error source-code" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--source-code-font-family); white-space: pre-wrap; font-size: var(--source-code-font-size) !important; line-height: 1.2;"><span style="box-sizing: border-box; min-width: 0px; min-height: 0px;">LinkError: WebAssembly.instantiate(): Import #92 module="env" function="time": imported function does not match the expected type</span></span></span></span></div><div class="" role="group" style="box-sizing: border-box; min-width: 0px; min-height: 0px; flex: 0 0 auto;"><span class="monospace stack-preview-container" style="display: inline-block; width: 1851px; --monospace-font-size: 11px; --monospace-font-family: dejavu sans mono,monospace; --source-code-font-size: 11px; --source-code-font-family: dejavu sans mono,monospace; box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--monospace-font-family); font-size: var(--monospace-font-size) !important;">
  | deserialize | @ | 870.1260b274e8c70c25…0b274e8c70c25c3c4:2
-- | -- | -- | --
  | S | @ | 870.1260b274e8c70c25…0b274e8c70c25c3c4:2
  | Promise.then (async) |   |  
  | apply | @ | 870.1260b274e8c70c25…0b274e8c70c25c3c4:2
  | h | @ | 413.df2804f5ae3431aa…804f5ae3431aa4c17:1
  | create | @ | 413.df2804f5ae3431aa…804f5ae3431aa4c17:1
  | startNew | @ | kernels.js:123
  | startNew | @ | sessions.js:105
  | (anonymous) | @ | index.js:366
  | route | @ | router.js:85
  | await in route (async) |   |  
  | fetch | @

ia-expanded="true" tabindex="-1" style="box-sizing: border-box; min-width: 0px; min-height: 0px; flex: 0 0 auto;"><span class="source-code" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--source-code-font-family); white-space: pre-wrap; font-size: var(--source-code-font-size) !important; line-height: 1.2;"><span class="console-message-anchor" style="box-sizing: border-box; min-width: 0px; min-height: 0px; float: right; text-align: right; max-width: 100%; margin-left: 4px;"><span class="devtools-link" role="link" tabindex="-1" title="http://localhost:8000/jupyter/extensions/@jupyterlite/xeus/static/870.1260b274e8c70c25c3c4.js?v=1260b274e8c70c25c3c4:2" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--sys-color-primary); text-decoration: underline; outline-offset: 2px; cursor: pointer; word-break: break-all;"><br class="Apple-interchange-newline">870.1260b274e8c70c25<span class="devtools-link-ellipsis" style="box-sizing: border-box; min-width: 0px; min-height: 0px;">…</span>0b274e8c70c25c3c4:2</span> </span><span class="console-message-text" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--override-error-text-color) !important;">Uncaught (in promise) <span class="object-value-error source-code" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--source-code-font-family); white-space: pre-wrap; font-size: var(--source-code-font-size) !important; line-height: 1.2;"><span style="box-sizing: border-box; min-width: 0px; min-height: 0px;">LinkError: WebAssembly.instantiate(): Import #92 module="env" function="time": imported function does not match the expected type</span></span></span></span></div><div class="" role="group" style="box-sizing: border-box; min-width: 0px; min-height: 0px; flex: 0 0 auto;"><span class="monospace stack-preview-container" style="display: inline-block; width: 1851px; --monospace-font-size: 11px; --monospace-font-family: dejavu sans mono,monospace; --source-code-font-size: 11px; --source-code-font-family: dejavu sans mono,monospace; box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--monospace-font-family); font-size: var(--monospace-font-size) !important;"><table class="stack-preview-container" style="box-sizing: border-box; min-width: 0px; min-height: 0px; display: block; width: 1851px;"><tr style="box-sizing: border-box; min-width: 0px; min-height: 0px; height: 16px; line-height: 16px;"><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;"></td><td class="function-name" style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px; max-width: 80em;">deserialize</td><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;">@</td><td class="link" style="box-sizing: border-box; min-width: 0px; min-height: 0px; cursor: pointer; text-decoration: underline; color: var(--sys-color-primary); outline-offset: 2px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px; width: 1676.67px; max-width: 0px;"><span class="devtools-link" role="link" tabindex="-1" jslog="Link; track: click; context: stack-trace-link" title="http://localhost:8000/jupyter/extensions/@jupyterlite/xeus/static/870.1260b274e8c70c25c3c4.js?v=1260b274e8c70c25c3c4:2" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--text-link); text-decoration: underline; outline-offset: 2px; cursor: pointer;">870.1260b274e8c70c25<span class="devtools-link-ellipsis" style="box-sizing: border-box; min-width: 0px; min-height: 0px;">…</span>0b274e8c70c25c3c4:2</span></td></tr><tr style="box-sizing: border-box; min-width: 0px; min-height: 0px; height: 16px; line-height: 16px;"><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;"></td><td class="function-name" style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px; max-width: 80em;">S</td><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;">@</td><td class="link" style="box-sizing: border-box; min-width: 0px; min-height: 0px; cursor: pointer; text-decoration: underline; color: var(--sys-color-primary); outline-offset: 2px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px; width: 1676.67px; max-width: 0px;"><span class="devtools-link" role="link" tabindex="-1" jslog="Link; track: click; context: stack-trace-link" title="http://localhost:8000/jupyter/extensions/@jupyterlite/xeus/static/870.1260b274e8c70c25c3c4.js?v=1260b274e8c70c25c3c4:2" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--text-link); text-decoration: underline; outline-offset: 2px; cursor: pointer;">870.1260b274e8c70c25<span class="devtools-link-ellipsis" style="box-sizing: border-box; min-width: 0px; min-height: 0px;">…</span>0b274e8c70c25c3c4:2</span></td></tr><tr style="box-sizing: border-box; min-width: 0px; min-height: 0px; height: 16px; line-height: 16px;"><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;"></td><td class="stack-preview-async-description" style="box-sizing: border-box; min-width: 0px; min-height: 0px; padding: 3px 0px 1px; font-style: italic; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;">Promise.then (async)</td><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;"></td><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;"></td></tr><tr style="box-sizing: border-box; min-width: 0px; min-height: 0px; height: 16px; line-height: 16px;"><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;"></td><td class="function-name" style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px; max-width: 80em;">apply</td><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;">@</td><td class="link" style="box-sizing: border-box; min-width: 0px; min-height: 0px; cursor: pointer; text-decoration: underline; color: var(--sys-color-primary); outline-offset: 2px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px; width: 1676.67px; max-width: 0px;"><span class="devtools-link" role="link" tabindex="-1" jslog="Link; track: click; context: stack-trace-link" title="http://localhost:8000/jupyter/extensions/@jupyterlite/xeus/static/870.1260b274e8c70c25c3c4.js?v=1260b274e8c70c25c3c4:2" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--text-link); text-decoration: underline; outline-offset: 2px; cursor: pointer;">870.1260b274e8c70c25<span class="devtools-link-ellipsis" style="box-sizing: border-box; min-width: 0px; min-height: 0px;">…</span>0b274e8c70c25c3c4:2</span></td></tr><tr style="box-sizing: border-box; min-width: 0px; min-height: 0px; height: 16px; line-height: 16px;"><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;"></td><td class="function-name" style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px; max-width: 80em;">h</td><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;">@</td><td class="link" style="box-sizing: border-box; min-width: 0px; min-height: 0px; cursor: pointer; text-decoration: underline; color: var(--sys-color-primary); outline-offset: 2px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px; width: 1676.67px; max-width: 0px;"><span class="devtools-link" role="link" tabindex="-1" jslog="Link; track: click; context: stack-trace-link" title="http://localhost:8000/jupyter/extensions/@jupyterlite/xeus/static/413.df2804f5ae3431aa4c17.js?v=df2804f5ae3431aa4c17:1" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--text-link); text-decoration: underline; outline-offset: 2px; cursor: pointer;">413.df2804f5ae3431aa<span class="devtools-link-ellipsis" style="box-sizing: border-box; min-width: 0px; min-height: 0px;">…</span>804f5ae3431aa4c17:1</span></td></tr><tr style="box-sizing: border-box; min-width: 0px; min-height: 0px; height: 16px; line-height: 16px;"><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;"></td><td class="function-name" style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px; max-width: 80em;">create</td><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;">@</td><td class="link" style="box-sizing: border-box; min-width: 0px; min-height: 0px; cursor: pointer; text-decoration: underline; color: var(--sys-color-primary); outline-offset: 2px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px; width: 1676.67px; max-width: 0px;"><span class="devtools-link" role="link" tabindex="-1" jslog="Link; track: click; context: stack-trace-link" title="http://localhost:8000/jupyter/extensions/@jupyterlite/xeus/static/413.df2804f5ae3431aa4c17.js?v=df2804f5ae3431aa4c17:1" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--text-link); text-decoration: underline; outline-offset: 2px; cursor: pointer;">413.df2804f5ae3431aa<span class="devtools-link-ellipsis" style="box-sizing: border-box; min-width: 0px; min-height: 0px;">…</span>804f5ae3431aa4c17:1</span></td></tr><tr style="box-sizing: border-box; min-width: 0px; min-height: 0px; height: 16px; line-height: 16px;"><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;"></td><td class="function-name" style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px; max-width: 80em;">startNew</td><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;">@</td><td class="link" style="box-sizing: border-box; min-width: 0px; min-height: 0px; cursor: pointer; text-decoration: underline; color: var(--sys-color-primary); outline-offset: 2px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px; width: 1676.67px; max-width: 0px;"><span class="devtools-link" role="link" tabindex="-1" jslog="Link; track: click; context: stack-trace-link" title="webpack://_JUPYTERLAB.CORE_OUTPUT/packages/kernel/lib/kernels.js:123" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--text-link); text-decoration: underline; outline-offset: 2px; cursor: pointer;">kernels.js:123</span></td></tr><tr style="box-sizing: border-box; min-width: 0px; min-height: 0px; height: 16px; line-height: 16px;"><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;"></td><td class="function-name" style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px; max-width: 80em;">startNew</td><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;">@</td><td class="link" style="box-sizing: border-box; min-width: 0px; min-height: 0px; cursor: pointer; text-decoration: underline; color: var(--sys-color-primary); outline-offset: 2px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px; width: 1676.67px; max-width: 0px;"><span class="devtools-link" role="link" tabindex="-1" jslog="Link; track: click; context: stack-trace-link" title="webpack://_JUPYTERLAB.CORE_OUTPUT/packages/session/lib/sessions.js:105" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--text-link); text-decoration: underline; outline-offset: 2px; cursor: pointer;">sessions.js:105</span></td></tr><tr style="box-sizing: border-box; min-width: 0px; min-height: 0px; height: 16px; line-height: 16px;"><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;"></td><td class="function-name" style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px; max-width: 80em;">(anonymous)</td><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;">@</td><td class="link" style="box-sizing: border-box; min-width: 0px; min-height: 0px; cursor: pointer; text-decoration: underline; color: var(--sys-color-primary); outline-offset: 2px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px; width: 1676.67px; max-width: 0px;"><span class="devtools-link" role="link" tabindex="-1" jslog="Link; track: click; context: stack-trace-link" title="webpack://_JUPYTERLAB.CORE_OUTPUT/packages/server-extension/lib/index.js:366" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--text-link); text-decoration: underline; outline-offset: 2px; cursor: pointer;">index.js:366</span></td></tr><tr style="box-sizing: border-box; min-width: 0px; min-height: 0px; height: 16px; line-height: 16px;"><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;"></td><td class="function-name" style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px; max-width: 80em;">route</td><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;">@</td><td class="link" style="box-sizing: border-box; min-width: 0px; min-height: 0px; cursor: pointer; text-decoration: underline; color: var(--sys-color-primary); outline-offset: 2px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px; width: 1676.67px; max-width: 0px;"><span class="devtools-link" role="link" tabindex="-1" jslog="Link; track: click; context: stack-trace-link" title="webpack://_JUPYTERLAB.CORE_OUTPUT/packages/server/lib/router.js:85" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--text-link); text-decoration: underline; outline-offset: 2px; cursor: pointer;">router.js:85</span></td></tr><tr style="box-sizing: border-box; min-width: 0px; min-height: 0px; height: 16px; line-height: 16px;"><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;"></td><td class="stack-preview-async-description" style="box-sizing: border-box; min-width: 0px; min-height: 0px; padding: 3px 0px 1px; font-style: italic; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;">await in route (async)</td><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;"></td><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px;"></td></tr><tr style="box-sizing: border-box; min-width: 0px; min-height: 0px; height: 16px; line-height: 16px;"><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px; color: rgb(31, 31, 31); font-family: &quot;dejavu sans mono&quot;, monospace; font-size: 11px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(252, 235, 235); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"></td><td class="function-name" style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px; max-width: 80em; color: rgb(31, 31, 31); font-family: &quot;dejavu sans mono&quot;, monospace; font-size: 11px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(252, 235, 235); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">fetch</td><td style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis; padding-inline: 2px; color: rgb(31, 31, 31); font-family: &quot;dejavu sans mono&quot;, monospace; font-size: 11px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(252, 235, 235); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">@</td><br class="Apple-interchange-newline">
</pre>
</details>

**Comment by DerThorsten** [comment]
Hi, thanks for reporting this.

A few questions:
* Are you running this piece of code in an async function or directly in a cell?
* Also, can you check if there is some error in the console? 
* Is the deployment also on `localhost`? If the deployment of jupyterlite is **not** on local host this cannot work.

A suggestion would be to just use the `requests` package which should just work

**Comment by jvailius** [comment]
> * Are you running this piece of code in an async function or directly in a cell?
directly
> * Also, can you check if there is some error in the console?
I'll check - but it should be very easy to reproduce
> * Is the deployment also on `localhost`? If the deployment of jupyterlite is **not** on local host this cannot work.
localhost: same url as I am using in the request
> A suggestion would be to just use the `requests` package which should just work
OK :)

**Comment by jvailius** [comment]
It seems there are "two" errors - at least error messages but the result is OK.

Even with requests I get the error mentioned above in the cell output if I wrap requests in an async function and await it:

> > 
> > AttributeError                            Traceback (most recent call last)
> > File /lib/python3.11/site-packages/IPython/__init__.py:44, in patch_asyncio.<locals>.asyncio_call(self, coro)
> >      41     task.add_done_callback(done_cb)
> >      43 raw_js_py_callback = pyjs.JsValue(py_callback)
> > ---> 44 js_py_callback = raw_js_py_callback['__usafe_void_val_val__'].bind(raw_js_py_callback)
> >      45 js_promise = pyjs.js.Promise.new(js_py_callback)
> >      47 pyjs.js.globalThis.toplevel_promise = js_promise
> > 
> > AttributeError: has no attribute/key __usafe_void_val_val__  
> 

Without async I just get an error in the browser console:

> Refused to set unsafe header "Accept-encoding"
> Refused to set unsafe header "Connection"

I am using async function sin a library because I use micropip and I'd like to make my library work with both kernels.

**Comment by DerThorsten** [comment]
I can confirm that toplevel awaits are not working atm
So as a temporary workaround, just wrap your cell in an async function.

Ie:
```
async def foo():
    # your code here
```

**Comment by DerThorsten** [comment]
I will probably be able to add a fix in the next days

**Comment by jvailius** [comment]
Thank you @DerThorsten. OT: I have tested xeus vs pyodide and I hope I can keep xeus. Being able to just provide the environment without extra installs is very useful.

**Comment by DerThorsten** [comment]
this is fixed (a patch in IPython needed to be updated)

To use this, you will need the most recent build of IPython (ipython=8.22.2=py311had7285e_1).
(FYI, the most recent IPython also needs a very recent traitlets package )

This is the script I used to create  working build

```bash
WASM_ENV_NAME=my-env-name
WASM_ENV_PREFIX=$MAMBA_ROOT_PREFIX/envs/$WASM_ENV_NAME
micromamba create -n $WASM_ENV_NAME \
            --platform=emscripten-wasm32 \
            -c https://repo.mamba.pm/emscripten-forge \
            -c https://repo.mamba.pm/conda-forge \
            --yes \
            xeus-python "ipython=8.22.2=py311had7285e_1" "traitlets>=5.14.2"

jupyter lite build \
        --XeusAddon.prefix=$WASM_ENV_PREFIX 
```

**Comment by jvailius** [comment]
Thanks for the additional hints. I'll give it a try.

**Comment by DerThorsten** [comment]
The ipython build with the fix : `ipython-8.22.2-py311had7285e_2`

therefore the updated snipped is:
```bash
WASM_ENV_NAME=my-env-name
WASM_ENV_PREFIX=$MAMBA_ROOT_PREFIX/envs/$WASM_ENV_NAME
micromamba create -n $WASM_ENV_NAME \
            --platform=emscripten-wasm32 \
            -c https://repo.mamba.pm/emscripten-forge \
            -c https://repo.mamba.pm/conda-forge \
            --yes \
            xeus-python "ipython=8.22.2=py311had7285e_2" "traitlets>=5.14.2"

jupyter lite build \
        --XeusAddon.prefix=$WASM_ENV_PREFIX 
```
